base/static/base/js/jquery.track-everything.js

```diff
@@ -63,362 +29,572 @@
-        // const isNewTab = event.target === '_blank' || isMiddleClick;
-        if (href && !isNewTab && !(eventName === 'tab' && href && href.startsWith('#'))) {
-            event.preventDefault(); // delay navigation just slightly
-            setTimeout(() => {
-                window.location.href = href;
-            }, 200); // give GA time to fire
-        }
-    }
+                    }
+                    // Catalog VuFind Record
+                    else if (window.location.href.indexOf(LOCATIONS.VUFIND_RECORD) > -1) {
+                        params.event_subcategory = params.event_subcategory ||
+                            link.closest('.top-navbar, .navbar-header, .navbar-collapse') ? 'Header Navbar' :
+                            link.closest('.search.container.navbar') ? 'Search Operations' :
+                                link.closest('.breadcrumbs') ? 'Breadcrumbs' :
+                                    link.closest('#bookplates') ? "Bookplates" :
+                                        link.closest('.media-left') ? "Record Media" :
+                                            link.closest('.savedLists') ? "Record Saved in Lists" :
+                                                link.closest('.media-body, .bibToggle') ? "Record Metadata" :
+                                                    link.closest('.related__title, .record-tab.similar, .tab-pane.similar-tab') ? 'Record Similar Items' :
+                                                        link.closest('.record-tab.holdings, .tab-pane.holdings-tab') ? 'Record Holdings' :
+                                                            link.closest('.record-tab.description, .tab-pane.description-tab') ? 'Record Description' :
+                                                                link.closest('.record-tab.toc, .tab-pane.toc-tab') ? 'Record Table of Contents' :
+                                                                    link.closest('.record-tab.details, .tab-pane.details-tab') ? 'Record Staff View' :
+                                                                        link.closest('.action-toolbar') ? 'Action Toolbar' :
+                                                                            'id:' + link.closest('[id]').getAttribute('id') || "VuFind Record Widget";
+                        params.event_label = link.closest('.savedLists') ? 'Record Saved in List' :
+                            link.closest('.bibToggle') ? 'More Details' : // How did VSCode knew to predict the value 'More Details' here?.
+                                link.classList.contains('title') ? 'Title' :
+                                    link.classList.contains('result-author') ? 'Author' :
+                                        link.classList.contains('external') ? 'Holding' :
+                                            link.classList.contains('save-record') ? 'Save Record' :
+                                                params.event_label || 'Unknown';

```

Comment:
This is pretty fragile and it will break silently. I'm not sure if anything can be done about that, though. The nature of what you're trying to achieve here will likely be fragile by definition. I'm not sure how maintainable it will be, on that note. I guess there's no harm in trying though.

Reply:
Yeah, I agree. It's a difficult problem to solve. I appreciate keeping the template changes out of VuFind. We'll go with this for now and see how it works out.